src/driver/odoc_driver.ml

```diff
@@ -633,27 +636,37 @@ let run mode
   let () =
     Eio.Fiber.both
       (fun () ->
-        let all =
+        let units =
           let all = Util.StringMap.bindings all |> List.map snd in
           let dirs =
             let odocl_dir = Option.value odocl_dir ~default:odoc_dir in
             { Odoc_unit.odoc_dir; odocl_dir; index_dir; mld_dir }
           in
-          Odoc_units_of.packages ~dirs ~extra_libs_paths all
+          Odoc_units_of.packages ~dirs ~extra_paths all
         in
-        Compile.init_stats all;
-        let compiled = Compile.compile ?partial ~partial_dir:odoc_dir all in
-        let linked = Compile.link compiled in
-        let occurrence_file =
-          let output =
-            Fpath.( / ) odoc_dir "occurrences-all.odoc-occurrences"
-          in
-          let () = Odoc.count_occurrences ~input:[ odoc_dir ] ~output in
-          output
+        Compile.init_stats units;
+        let compiled =
+          match actions with
+          | LinkAndGen -> units
+          | _ -> Compile.compile ?partial ~partial_dir:odoc_dir units
```

Comment:
Even though we might not need to be extra future proof, an exhaustive check here is cheap and useful!